eventcore-testing/src/chaos.rs

```diff
@@ -4,13 +4,38 @@ use eventcore_types::{
     Event, EventStore, EventStoreError, EventStreamReader, EventStreamSlice, Operation, StreamId,
     StreamWrites,
 };
+use nutype::nutype;
 use rand::{Rng, SeedableRng, random, rngs::StdRng};
 
+/// Probability value for chaos engineering injection rates.
+///
+/// Probability represents a value in the range [0.0, 1.0] where 0.0 means
+/// never inject failures and 1.0 means always inject failures.
+///
+/// # Examples
+///
+/// ```ignore
+/// use eventcore_testing::chaos::Probability;
+///
+/// let never = Probability::try_new(0.0).unwrap();
+/// let sometimes = Probability::try_new(0.5).unwrap();
+/// let always = Probability::try_new(1.0).unwrap();
+///
+/// // Values outside [0.0, 1.0] are rejected
+/// assert!(Probability::try_new(1.5).is_err());
+/// assert!(Probability::try_new(-0.1).is_err());
+/// ```
+#[nutype(
+    validate(greater_or_equal = 0.0, less_or_equal = 1.0),
+    derive(Debug, Clone, Copy, PartialEq, PartialOrd, Display, Into)
+)]
+pub struct Probability(f32);
+
 #[derive(Debug, Clone)]
 pub struct ChaosConfig {
     deterministic_seed: Option<u64>,
-    failure_probability: f32,
-    version_conflict_probability: f32,
+    failure_probability: Probability,
+    version_conflict_probability: Probability,
```

Comment:
Should these each be different domain types, so that there is no way to inadvertently use the wrong one?